src/renderer/components/NodeSelectorPanel/NodeSelectorPanel.tsx

```diff
@@ -102,123 +107,188 @@ const NodeSelector = memo(({ schemata, height }: NodeSelectorProps) => {
         return [...byCategories.values()].flat().filter((n) => favorites.has(n.schemaId));
     }, [byCategories, favorites]);
 
+    const [showCollapseButtons, setShowCollapseButtons] = useState(false);
+
+    const defaultIndex = [['', favoriteNodes]].concat([...byCategories]).map((_, index) => index);
```

Comment:
This is an array with the number 0 to `byCategories.size` (inclusive).
```suggestion
    const defaultIndex = Array.from({ length: byCategories + 1 }, (_, i) => i)
```

Reply:
This fix throws up an error about trying to add a Map<> to a number. Is this the change you had in mind?
```suggestion
    const defaultIndex = Array.from({ length: [...byCategories].length + 1 }, (_, i) => i);
```

Reply:
My bad, it has to be `byCategories.size`. I fixed my suggestion.